enhancements/oc/cli-manager.md

```diff
@@ -0,0 +1,252 @@
+  * Anyone who wants such tools can download them outside of this mechanism
+* CRD controller for registering CRs with an API for listing, extracting and downloading artifacts
+  * Krew-compatible index that both krew and oc can consume where the download links are for cluster-local resources
+  * The index and artifcats will served from a single route and service within the cluster
+
+### Non-Goals
+
+* `oc tools` will not build or serve the binaries.  It will know where to find them.
+* This proposal is not concerned with _which_ binaries will be managed.  This proposal is meant to determine the mechanism only.  Consumers and publishers are clients of the mechanism.
+* `oc tools` will not create or update the ClusterCLI Custom Resources, that will be managed by individual CLIs. However, the CLIs that are
+currently available as `consoleclidownloads`, `oc, odo, and helm`, will be included in the reference implementation of this enhancement.
+
+## Proposal
+
+Each component wishing to provide customers with their binaries will build and publish artifacts via an official channel to a central index.
+Each component wishing to provide customers with binaries will create a ClusterCLI custom resource to provide an image name and the file path
+within that image for the binaries. The CLIs currently offered via console-cli-downloads will be included in the reference implementation of this enhancement.
+`oc` will gather ClusterCLI from the cluster, and also the fileserver will offer the binaries via the console by mounting the binaries
+from the location/image in each custom resource.
+
+CLIs could be installed in a similar manner to plugins through [krew](https://github.com/kubernetes-sigs/krew).
+CLI images would serve the artifacts to enable disconnected downloads, and OpenShift CLIs will be available as
+[krew plugins](https://github.com/kubernetes-sigs/krew-index/tree/master/plugins).
+
+* Krew and Krew plugins are upstream projects that Kubernetes users are already familiar with
+* The index will be served by the CRD controller with its contents managed by cluster-admins via CRs
+* A Krew-compatible index can provide available CLIs and plugins for a cluster (similar to the current index) and can also provide an easy mechanism for installing through `oc tools`
+* An openshift-krew-index with supported artifacts could hold information about CLIs supported by OpenShift.
+* With `krew index add https://someother-third-party-index` we won't limit cluster-admins from adding their own index with whatever plugins they want. Only the openshift-krew-index will be supported.
+
+Users will continue to have the option to download binaries from either the central location where artifacts are published or from the console.
+This proposal adds the mechanism to install from images built from the same artifacts using `oc`. All options will offer the same binaries,
+ensured through sha256 signatures.
```

Comment:
sha256 is a hash/a checksum, I don't think it should/can be called a signature. "SHA256 hashes" is used later in the document.

Reply:
Updated. Thank you.